test/e2e-go/cli/goal/expect/statefulTealCreateAppTest.exp

```diff
@@ -88,7 +88,7 @@ proc statefulTealTest { TEST_ALGO_DIR TEST_DATA_DIR TEAL_PROGRAM} {
             "Please enter the password for wallet '$WALLET_1_NAME':" {send "$WALLET_1_PASSWORD\r" ; exp_continue }
             "*not opted in to app*" {puts "received expected error"; close}
             "*exceeds schema bytes count*" {puts "received expected error"; close}
-            "Couldn't broadcast tx with algod: HTTP 400 Bad Request: TransactionPool.Remember: transaction*" {puts "received expected error"; close}
+            "Couldn't broadcast tx with algod: HTTP 400 Bad Request: *TransactionPool.Remember: transaction*" {puts "received expected error"; close}
```

Comment:
It's because now we return a Data field in the json for these errors.  The existing restClient.go::extractError function returns _just_ the message if there is no Data field.  But if there is, it returns the JSON string --- it's a lot like exception coercion I found in the py sdk.  Therefore `goal` ends up printing the json response, including all the structure info, instead of just the message.

I can see an argument for trying to keep the message exactly the same, in case of scripting use.  But I think it's better to return the extra stuff (this is what must happen when other endpoints that use Data are called too).  It's useful information, contains at least what was always there, and scripts are more likely to care only about success/failure, not the exact detail of failure.  If there _is_ code parsing the existing message closely, I bet it's some attempt to provide debugging info, and I'd very much like them to notice the new info.

Reply:
An interesting alternative would be for me to remarshall the JSON back into a `baiscs.SError` in such a way that the message would be the same as it was before, but a sufficiently interested client could interrogate the attributes.  If I did that, the next question is, should `goal` be printing the attributes?

Reply:
Of course, the downside to _that_ is that any existing users of `extractError` that _wanted_ the json to appear in the error message now have to be changed to look inside the error for it.

Reply:
Recording here that we decide offline to change extractError to stop embedding the JSON in the message when `.Data` is populated and instead stash the actual JSON `map[string]any` in a field of the returned HTTPError (a bit simpler than creating a `basics.SError`)

Reply:
Is the plan to modify extractError in a different PR?

Reply:
Nah, I want to do it here, so I haven't resolved this convo.

Reply:
Resolving now, as I've committed the change, though it's likely to trip up some e2e test or another.